dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1057,4 +1059,18 @@ private void CaptureUsageDetails(CompletionsUsage usage)
         s_completionTokensCounter.Add(usage.CompletionTokens);
         s_totalTokensCounter.Add(usage.TotalTokens);
     }
+
+    private async Task<BinaryData?> GetBinaryDataFromAudioContentAsync(
+        AudioContent audioContent,
+        CancellationToken cancellationToken)
+    {
+        BinaryData? data = audioContent?.Data;
+
+        if (data is null && audioContent?.Stream is not null)
```

Comment:
@SergeyMenshykh it's warning in this case.
<img width="415" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/13853051/ec20635f-2924-4a2b-afcd-67b98fca10c0">

Most probably it doesn't know whether `data` is null because `audioContent` is null or because `audioContent.Data` is null, so it requires additional check on compile-time.